Public/Src/Engine/Scheduler/IncrementalScheduling/GraphAgnosticIncrementalSchedulingState.cs

```diff
@@ -649,13 +658,43 @@ private void ProcessDynamicallyObservedChangedPath(string changedPathStr, PathCh
             }
 
             // Determine the set to use based on the change (Enumeration or Files)
-            var observationType = (changeReasons & PathChanges.MembershipChanged) != 0 
-                ? DynamicObservationType.Enumeration 
-                : DynamicObservationType.ObservedFile;
-            var dynamicObservationsToInvalidate = observationType == DynamicObservationType.Enumeration 
-                ? m_dynamicallyObservedEnumerations 
-                : m_dynamicallyObservedFiles;
+            if ((changeReasons & PathChanges.MembershipChanged) != 0)
+            {
+                ProcessDynamicallyObservedChangedPath(
+                    changedPathStr, 
+                    changeReasons, 
+                    path, 
+                    DynamicObservationType.Enumeration, 
+                    m_dynamicallyObservedEnumerations);
+            }
+            else
+            {
+                if ((changeReasons & PathChanges.Removed) != 0)
+                {
+                    ProcessDynamicallyObservedChangedPath(
+                        changedPathStr, 
+                        changeReasons, 
+                        path, 
+                        DynamicObservationType.ProbedFile, 
+                        m_dynamicallyProbedFiles);
+                }

```

Comment:
should created also go through this path?

Reply:
It's only for existing file probe. If the file is recreated, then the resulting probe will be the same.

---
In reply to: [344813695](https://github.com/microsoft/BuildXL/pull/1189#discussion_r344813695) [](ancestors = 344813695)